### PR TITLE
feat(apple): Filter app hang events

### PR DESCRIPTION
**Pull request by philipphofmann**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Explain how to filter app hang events in beforeSend.

### CODE REVIEW

File: docs/platforms/apple/common/configuration/app-hangs.mdx [hunk 1]
```diff
@@ -18,6 +18,38 @@ The SDK reports an app hang immediately, but doesn’t report the exact duration
...
+
+<SignInNote />
+
+```swift {tabTitle:Swift}
+import Sentry
+
+SentrySDK.start { options in
+    options.dsn = "___PUBLIC_DSN___"
+    options.beforeSend = { event in
+        if (event.exceptions?.first?.type == "App Hanging") {
```
Comment (kahest) [inline docs/platforms/apple/common/configuration/app-hangs.mdx:31]: `l`: is this preferred to checking mechanism , e.g `if event.exceptions?.contains(where: { $0.mechanism?.type = "AppHang" }) {`?

**Review by kahest — APPROVED** [review]
thanks for adding this! just left a question for my own education :)

File: docs/platforms/apple/common/configuration/app-hangs.mdx [hunk 2]
```diff
@@ -18,6 +18,38 @@ The SDK reports an app hang immediately, but doesn’t report the exact duration
 
 Because the app hang detection integration uses SentryCrashIntegration to capture the stack trace when creating app hang events, SentryCrashIntegration has to be enabled for the integration to work.
 
+You can filter app hang events in beforeSend by checking the event exception type:
```
Comment (vivianyentran) [inline docs/platforms/apple/common/configuration/app-hangs.mdx:21]:
```suggestion
You can filter and modify app hang events in `beforeSend` by checking the event exception type:
```

**Review by vivianyentran — APPROVED** [review]

File: docs/platforms/apple/common/configuration/app-hangs.mdx (hunk 1 above)
Comment (philipphofmann) [reply docs/platforms/apple/common/configuration/app-hangs.mdx:31]: Right now yes. We could also consider making isAppHangEvent of the Event.m public https://github.com/getsentry/sentry-cocoa/blob/aa45f3606be62823e6f0fd8f0d6a704b16867e17/Sources/Sentry/SentryEvent.m#L203-L207.

**Pull request merged**